# sarek-ngs-pipeline
From FASTQ to the analysis-ready variant-containing TSV file 


In this tutorial, starting from the raw sequencing data, I will work you through some simple steps to get the final list of SNPs and Indels in the samples with detailed annotation. To do this, we will take advantage of the nf-core/sarek pipeline.


## Step 1: Run the nf-core/sarek pipeline
We are going to use the RLS sequencing data generated by Illumina platform.

First, we will need to prepare a sample.tsv file containing the following columns: subject, sex, status, sample, lane, path/to/the/fastq1, path/to/the/fastq2.

For example: 

|   |   |   |   |   |   |   |
| ------- | -- | - | -------- | - | ------------------ | ------------------ |
| Path083 | xx | 0 | sample01 | 1 | Path083_1.fastq.gz | Path083_2.fastq.gz |
| Path0187 |	xx | 0 | sample02 | 2 | Path187_1.fastq.gz | Path187_2.fastq.gz |
| Path248	| xx | 0 | sample03 | 3 | Path248_1.fastq.gz | Path248_2.fastq.gz |

Next, make a nextflow.config file containing your custom configurations. In our case, we want to call SNPs and Indels so we are going to use HaplotypeCaller tool. We also want to monitor our run on the nextflow tower so we will add the tower details. 

	params {
		input = 'sample.tsv'
		genome = 'GRCh38'
		tools = 'HaplotypeCaller'  
		}
	tower {
		accessToken = '<<add your token>>'
		endpoint = 'https://nftower.qut.edu.au/api'
		enabled = true
		}
    
Finally, the launch.pbs file which will look like the followings:

```
  #!/bin/bash -l
  #PBS -N sarek-haplotypecaller
  #PBS -l walltime=168:00:00
  #PBS -l nodes=2:ppn=16,mem=128gb
  #PBS -m abe
  #PBS -M <<your email>>
  cd $PBS_O_WORKDIR
  conda activate nf-env
  nextflow run nf-core/sarek -profile singularity 
```






    
    
